### PR TITLE
Fix: When the "Optimized DOM" experiment is off and custom breakpoints are defined... [ED-4994]

### LINKED ISSUE #16322: Section > Responsive > Reverse not work for mobile, if I change the breakpoint.
## Prerequisites

- [ ] I have searched for similar issues in both open and closed tickets and cannot find a duplicate.
- [ ] The issue still exists against the latest stable version of Elementor.

## Isolating the problem

- [ ] This bug happens with only Elementor plugin active (and Elementor Pro).
- [ ] This bug happens with a default WordPress theme active.
- [ ] I can reproduce this bug consistently using the steps above.

## Environment

<details>
<summary>System Info</summary>
```
<!-- For bug reports, let us know about your system environment: Copy and paste the system info report from Elementor => System info, and paste it here or in http://pastebin.com/ -->
```
</details>

### PR DESCRIPTION
**Pull request by udidol**
…, some responsive CSS receives default breakpoint values.

## PR Type
What kind of change does this PR introduce?

- [X] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

* Fix: When the "Optimized DOM" experiment is off and custom breakpoints are defined, some responsive CSS receives default breakpoint values. Fixes #16322

### CODE REVIEW

File: includes/frontend.php [hunk 1]
```diff
@@ -681,6 +681,26 @@ public function enqueue_styles() {
 		}
 	}
 
+	/**
+	 * Get Custom Frontend File URL
+	 *
+	 * Generate a custom frontend CSS file based on a passed template file name, and return the generated file's URL.
+	 *
+	 * @since 3.4.5
+	 * @access public
```
Comment (rotemee) [inline includes/frontend.php:690]: access = private.

File: includes/frontend.php [hunk 2]
```diff
@@ -552,9 +544,17 @@ public function register_styles() {
 
 		if ( ! Plugin::$instance->experiments->is_feature_active( 'e_dom_optimization' ) ) {
 			// If The Dom Optimization feature is disabled, register the legacy CSS
+			$frontend_legacy_file_name = 'frontend-legacy' . $direction_suffix . $min_suffix . '.css';
+
+			if ( $has_custom_file ) {
+				$frontend_legacy_file_url = $this->get_custom_frontend_file_url( $frontend_legacy_file_name );
+			} else {
+				$frontend_legacy_file_url = ELEMENTOR_ASSETS_URL . 'css/' . $frontend_legacy_file_name;
+			}
```
Comment (rotemee) [inline includes/frontend.php:553]: Maybe better to move it to the same function that generates the file URL and the logic will be part of that function?

File: includes/frontend.php (hunk 1 above)
Comment (udidol) [reply includes/frontend.php]: Done

File: includes/frontend.php (hunk 2 above)
Comment (udidol) [reply includes/frontend.php]: Done

File: includes/frontend.php [hunk 3]
```diff
@@ -532,29 +543,15 @@ public function register_styles() {
...
-			}
-
-			$frontend_file_url = $frontend_file->get_url();
-		} else {
-			$frontend_file_url = ELEMENTOR_ASSETS_URL . 'css/' . $frontend_file_name;
-		}
-
 		$frontend_dependencies = [];
 
+		$this->has_custom_breakpoints = Plugin::$instance->breakpoints->has_custom_breakpoints();
```
Comment (kobizz) [inline includes/frontend.php:548]: Make it a local variable

**Review by kobizz — CHANGES_REQUESTED** [review]

**Review by kobizz — APPROVED** [review]

**Pull request merged**